test/unit/BPool.t.sol

```diff
@@ -702,32 +702,134 @@ contract BPool_Unit_ExitPool is BasePoolTest {
   }
 
-  function test_Revert_NotFinalized() private view {}
+  function test_Revert_NotFinalized(ExitPool_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    _setFinalize(false);
 
-  function test_Revert_MathApprox() private view {}
+    vm.expectRevert('ERR_NOT_FINALIZED');
+    bPool.exitPool(_fuzz.poolAmountIn, _zeroAmountsArray());
+  }
 
-  function test_Pull_PoolShare() private view {}
+  function test_Revert_MathApprox(ExitPool_FuzzScenario memory _fuzz, uint256 _poolAmountIn) public happyPath(_fuzz) {
+    _poolAmountIn = bound(_poolAmountIn, 0, INIT_POOL_SUPPLY / 2 / BONE); // bdiv rounds up
 
-  function test_Push_PoolShare() private view {}
+    vm.expectRevert('ERR_MATH_APPROX');
+    bPool.exitPool(_poolAmountIn, _zeroAmountsArray());
+  }
 
-  function test_Burn_PoolShare() private view {}
+  function test_Pull_PoolShare(ExitPool_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    assertEq(bPool.balanceOf(address(this)), _fuzz.initPoolSupply);
+
+    bPool.exitPool(_fuzz.poolAmountIn, _zeroAmountsArray());
 
-  function test_Revert_TokenArrayMathApprox() private view {}
+    assertEq(bPool.balanceOf(address(this)), _fuzz.initPoolSupply - _fuzz.poolAmountIn);
+  }
 
-  function test_Revert_TokenArrayLimitOut() private view {}
+  function test_Push_PoolShare(ExitPool_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    uint256 _exitFee = bmul(_fuzz.poolAmountIn, EXIT_FEE);
+    uint256 _balanceBefore = bPool.balanceOf(bPool.call__factory());
 
-  function test_Revert_Reentrancy() private view {}
+    bPool.exitPool(_fuzz.poolAmountIn, _zeroAmountsArray());
 
-  function test_Set_TokenArrayBalance() private view {}
+    assertEq(bPool.balanceOf(bPool.call__factory()), _balanceBefore - _fuzz.poolAmountIn + _exitFee);
+  }
 
-  function test_Emit_TokenArrayLogExit() private view {}
+  function test_Burn_PoolShare(ExitPool_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    uint256 _exitFee = bmul(_fuzz.poolAmountIn, EXIT_FEE);
+    uint256 _pAiAfterExitFee = bsub(_fuzz.poolAmountIn, _exitFee);
+    uint256 _totalSupplyBefore = bPool.totalSupply();
 
-  function test_Push_TokenArrayTokenAmountOut() private view {}
+    bPool.exitPool(_fuzz.poolAmountIn, _zeroAmountsArray());
 
-  function test_Emit_LogCall() private view {}
+    assertEq(bPool.totalSupply(), _totalSupplyBefore - _pAiAfterExitFee);
+  }
+
+  function test_Revert_TokenArrayMathApprox(
+    ExitPool_FuzzScenario memory _fuzz,
+    uint256 _tokenIndex
+  ) public happyPath(_fuzz) {
+    _assumeHappyPath(_fuzz);
+    _tokenIndex = bound(_tokenIndex, 0, TOKENS_AMOUNT - 1);
+    _fuzz.balance[_tokenIndex] = 0;
+    _setValues(_fuzz);
+
+    vm.expectRevert('ERR_MATH_APPROX');
+    bPool.exitPool(_fuzz.poolAmountIn, _zeroAmountsArray());
+  }
+
+  function test_Revert_TokenArrayLimitOut(ExitPool_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    vm.expectRevert('ERR_LIMIT_OUT');
+    bPool.exitPool(_fuzz.poolAmountIn, _maxAmountsArray());
+  }
```

Comment:
hmmm ok for now i'd say, but it'd be interesting to see that calculating the `balancesOut`, we can send it and won't revert, but if we ++ any value in the array, it'll revert

this expectation is too broad

Reply:
Check now